Guide/form.markdown

```diff
@@ -5,8 +5,346 @@
+
+### Help Texts
+
+You can add a help text below a form control like this:
+
+```html
+{(textField #title) { helpText = "Max. 140 characters"} }
+```
+
+This will render like:
+
+```html
+<div class="form-group" id="form-group-post_title">
+    <label for="post_title">Title</label>
+
+    <input type="text" name="title" id="post_title" class="form-control">
+    <small class="form-text text-muted">Max. 140 characters</small>
+</div>
+```
+
+### Custom Field Label Text
+
+By default the field name will be used as a label text. The camel case field name will be made more human readable of course, so `contactName` will turn to `Contact Name`, etc. Sometimes you want to change this auto-generated input label to something custom. Use `fieldLabel` for that, like this:
+
+```html
+{(textField #title) { fieldLabel = "Post Title"} }
+```
+
+This will render like:
+
+```html
+<div class="form-group" id="form-group-post_title">
+    <label for="post_title">Post Title</label>
+    <input type="text" name="title" id="post_title" class="form-control">
+</div>
+```
+
+### Custom CSS Classes
+
+You can add custom CSS classes to the input and label for better styling. Set `fieldClass` for adding a class to the input element and `labelClass` for the label element:
+
+```html
+{(textField #title) { fieldClass="title-input", labelClass = "title-label" } }
+```
+
+This will render like:
+
+```html
+<div class="form-group" id="form-group-post_title">
+    <label class="title-label" for="post_title">Title</label>
+    <input type="text" name="title" id="post_title" class="form-control title-input">
+</div>
+```
+
+Of course, the CSS classes for validation are still set as expected.
+
+### Placeholder
+
+You can specify an input placeholder like this:
+
+```html
+{(textField #title) { placeholder = "Enter your title ..." } }
+```
+
+This will render like:
+
+```html
+<div class="form-group" id="form-group-post_title">
+    <label for="post_title">Title</label>
+
+    <input type="text" name="title" id="post_title" placeholder="Enter your title ..." class="form-control">
+</div>
+```
+
+### Custom Submit Button Text
+
+Customize it like this:
+
+```html
+{submitButton { label = "Create it!" } }
+```
+
+This will render like:
+
+```html
+<button class="btn btn-primary">Create it!</button>
+```
+
+When you want to use e.g. an icon inside your button, it might be easier to just write the HTML manually by hand then.
+
+### Custom Submit Button Class
+
+Customize it like this:
+
+```html
+{submitButton { buttonClass = "create-button" } }
+```
+
+This will render like:
+
+```html
+<button class="btn btn-primary create-button">Create Post</button>
+```
+
+
+### Advanced Customization Options
+
+The following options are not commonly used, but are useful sometimes.
+
+#### Adding custom attributes to the input element
+
+Form rendering is built on top of blaze html. So you need to import the blaze html functions for this. Add this at the top of your module:
```

Comment:
Add link to https://hackage.haskell.org/package/blaze-html

Reply:
Done